src/include/transaction/transaction_context.h

```diff
@@ -245,18 +67,18 @@ class TransactionContext {
    * @param slot the TupleSlot being updated
    * @return a persistent pointer to the head of a memory chunk large enough to hold the undo record
    */
-  storage::DeltaRecord *UndoRecordForInsert(storage::DataTable *table, const storage::BlockLayout &layout,
-                                            storage::TupleSlot slot) {
+  storage::UndoRecord *UndoRecordForInsert(storage::DataTable *table, const storage::BlockLayout &layout,
+                                           storage::TupleSlot slot) {
     // TODO(Tianyu): Remove magic constant
     // Pretty sure we want 1, the primary key column?
-    uint32_t size = storage::DeltaRecord::Size(layout, {1});
-    storage::DeltaRecord *result = undo_buffer_.NewEntry(size);
-    return storage::DeltaRecord::InitializeDeltaRecord(result, txn_id_, slot, table, layout, {1});
+    uint32_t size = storage::UndoRecord::Size(layout, {1});
+    storage::UndoRecord *result = undo_buffer_.NewEntry(size);
+    return storage::UndoRecord::InitializeDeltaRecord(result, txn_id_, slot, table, layout, {1});
   }
 
  private:
   const timestamp_t start_time_;
   timestamp_t txn_id_;
```

Comment:
This doesn't need to be atomic, right? No one else is really looking at it besides the TransactionManager and eventually GC.

Just confirming.

Reply:
Right